database/src/migrations/20230817000000_critter_observation_deployment_tables.ts

```diff
@@ -0,0 +1,184 @@
+  );
+
+  COMMENT ON COLUMN deployment.deployment_id            IS 'System generated surrogate primary key identifier.';
+  COMMENT ON COLUMN deployment.critter_id              IS 'The id of the critter. Internal Sims id.';
+  COMMENT ON COLUMN deployment.bctw_deployment_id       IS 'The external system id of a BCTW deployment / collar->animal assignment.';
+  COMMENT ON COLUMN deployment.create_date              IS 'The datetime the record was created.';
+  COMMENT ON COLUMN deployment.create_user              IS 'The id of the user who created the record as identified in the system user table.';
+  COMMENT ON COLUMN deployment.update_date              IS 'The datetime the record was updated.';
+  COMMENT ON COLUMN deployment.update_user              IS 'The id of the user who updated the record as identified in the system user table.';
+  COMMENT ON COLUMN deployment.revision_count          IS 'Revision count used for concurrency control.';
+  COMMENT ON TABLE  deployment                          IS 'Information about individual critters associated to specific bctw deployments.';
+
+  -- Add foreign key constraint from child table to parent table on critter_id
+  ALTER TABLE deployment ADD CONSTRAINT deployment_fk1
+    FOREIGN KEY (critter_id)
+    REFERENCES critter(critter_id);
+
+  -- Add unique constraint
+  CREATE UNIQUE INDEX deployment_uk1 ON deployment(critter_id, bctw_deployment_id);
+
+  -- Create audit and journal triggers
+  create trigger audit_critter before insert or update or delete on deployment for each row execute procedure tr_audit_trigger();
+  create trigger journal_critter after insert or update or delete on deployment for each row execute procedure tr_journal_trigger();
+
+
+  ----------------------------------------------------------------------------------------
+  -- Create Critter Observation count trigger
+  ----------------------------------------------------------------------------------------
+
+  CREATE OR REPLACE FUNCTION biohub.tr_critter_observation_count()
+
+   RETURNS trigger
+   LANGUAGE plpgsql
+  AS $function$
+    -- *******************************************************************
+
+    -- Procedure: tr_critter_observation_count
+    -- Purpose: Validates amount of critters of a observation is less than or equal to observation count.
+    --
+
+    -- MODIFICATION HISTORY
+    -- Person           Date        Comments
+    -- ---------------- ----------- --------------------------------------
+    -- mac.deluca@quartech.com
+    --                  2023-17-08  initial release
+    -- *******************************************************************
+    DECLARE
+      num_critters integer := (SELECT count(*) FROM critter WHERE critter.observation_id = NEW.observation_id);
+      observation_total_count integer := (SELECT total_count FROM observation WHERE observation.observation_id = NEW.observation_id);
+    BEGIN
+      IF (num_critters >= observation_total_count) THEN
```

Comment:
Should this be greater than (rather than greater or equal)?

Reply:
I have the trigger firing on before insert. So the count of critters would be the table before the new record is added.

Reply:
Do we want the trigger to fire on update aswell or just insert?

Reply:
Updated this to not fire on update. (It was incorrectly firing the trigger)